src/faidx/mod.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright 2019 Manuel Landesfeind, Evotec International GmbH
+// Licensed under the MIT license (http://opensource.org/licenses/MIT)
+// This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//!
+//! Module for working with faidx-indexed FASTA files.
+//!
+
+use std::ffi;
+use std::path::Path;
+use url::Url;
+
+use crate::htslib;
+
+pub mod errors;
+pub use errors::{Error, Result};
+
+fn path_as_bytes<'a, P: 'a + AsRef<Path>>(path: P, must_exist: bool) -> Result<Vec<u8>> {
+    if path.as_ref().exists() || !must_exist {
+        Ok(path
+            .as_ref()
+            .to_str()
+            .ok_or(Error::NonUnicodePath)?
+            .as_bytes()
+            .to_owned())
+    } else {
+        Err(Error::FileNotFound {
+            path: path.as_ref().to_owned(),
+        })
+    }
+}
```

Comment:
If this function might be useful for other `Reader`s, it could go into `utils.rs`?

Reply:
I agree. But I followed the general pattern of the lib. For example, there is also `bam/errors.rs` etc. If all maintainer agree, then we can start combining this.

Reply:
Jep, I guess generic errors such as `FileNotFound` and `NonUnicodePath` could be maybe be moved up to a new `src/errors.rs` to avoid redundancies.

And `path_as_bytes` could easily live alongside `path_to_cstring` in `utils.rs`:
https://github.com/rust-bio/rust-htslib/blob/61b0644b73ab40612cd23e104ff0a3fe647bf064/src/utils.rs#L30-L34

Reply:
Yes, I can do that but its a larger work-package (considering a harmonization with the other `**/errors.rs`) and maybe we want to separate this from the faidx implementation? How about merging as-is and creating a separate issue/enhancement with an according branch to work on the generalization of errors. If you are willing to review more pull-request from me, I am open to work on the new errors-issue :-)

Reply:
Jep, sounds like a good way forward. I create issue #215 for putting general errors in a new `src/errors.rs`. And I guess having that is then a prerequisite for putting `path_as_bytes` in `src/utils.rs`, as that is using exactly those errors. So I've added that into the new issue, as well. And then, let's move forward with #214.